From 6103fff146d9e434a643a031f7a557fd793c8d08 Mon Sep 17 00:00:00 2001
From: Chris Morin <cmtm@google.com>
Date: Mon, 10 Dec 2018 20:32:42 -0800
Subject: [PATCH 26/26] make tlsdated less noisy when cros-debug is set

tlsdated gets extremely noisy when it's built with the cros-debug flag.
Turn off these log messages by default. They can be turned back on by
passing 2 '-v' flags to tlsdated, or by setting 'verbose' to '2' in the
tlsdated.conf file.

Also, don't pass '-v' to tlsdated because it's already set in the
tlsdated.conf file.

BUG=none
TEST=Ensure logs aren't full of messages from tlsdated
TEST=Pass '-vv' to tlsdated and ensure extra logs are seen
TEST=Set verbose to 2 in the tlsdated.conf file and ensure extra logs
are seen

Change-Id: I83db724bf739ae2808edd13b81252e037c983faa
Reviewed-on: https://chromium-review.googlesource.com/1370630
Commit-Ready: Christopher Morin <cmtm@google.com>
Tested-by: Christopher Morin <cmtm@google.com>
Reviewed-by: Mike Frysinger <vapier@chromium.org>
---
 init/tlsdated.conf | 2 +-
 src/tlsdated.c     | 7 +++++--
 src/util.h         | 2 +-
 3 files changed, 7 insertions(+), 4 deletions(-)

Index: tlsdate/src/tlsdated.c
===================================================================
--- tlsdate.orig/src/tlsdated.c
+++ tlsdate/src/tlsdated.c
@@ -186,7 +186,7 @@ parse_argv (struct opts *opts, int argc,
           opts->should_save_disk = 0;
           break;
         case 'v':
-          verbose = 1;
+          verbose += 1;
           break;
         case 'b':
           verbose_debug = 1;
@@ -371,7 +371,10 @@ load_conf (struct opts *opts)
         }
       else if (!strcmp (e->key, "verbose"))
         {
-          verbose = e->value ? !strcmp (e->value, "yes") : 1;
+          if (e->value)
+            verbose = !strcmp(e->value, "yes") ? 1 : strtol(e->value, NULL, 0);
+          else
+            verbose = 1;
         }
       else if (!strcmp (e->key, "source"))
         {
